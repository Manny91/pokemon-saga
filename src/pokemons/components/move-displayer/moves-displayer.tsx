import React from "react";
import { Loading } from "../loading/loading";
import { PokemonMove } from "../../../services/pokemon.service";
import styled from "../../../styled-components";

type Props = {
  handleMoveNext: () => void;
  handleMovePrevious: () => void;
  move: PokemonMove;
  loadingMoves: boolean;
};
export function PokemonMoveDisplayer({
  handleMoveNext,
  handleMovePrevious,
  move,
  loadingMoves
}: Props) {
  function displayMovement(move: PokemonMove) {
    return (
      <>
        <MovementName>{move.name}</MovementName>
        <MovementStatList>
          <MovementStat>Power : {move.power || "None"}</MovementStat>
          <MovementStat>Accuracy : {move.accuracy}</MovementStat>
          <MovementStat>PP : {move.pp}</MovementStat>
        </MovementStatList>
      </>
    );
  }
  return (
    <>
      <MovesWrapper>
        <MoveInfoWrapper>
          {loadingMoves && <Loading />}
          {!loadingMoves && move && displayMovement(move)}
        </MoveInfoWrapper>
        <MoveTypeWrapper></MoveTypeWrapper>
      </MovesWrapper>

      <ButtonWrapper>
        <ButtonMove onClick={handleMovePrevious}>
          <i className="fas fa-caret-up" />
        </ButtonMove>
        <ButtonMove onClick={handleMoveNext}>
          <i className="fas fa-caret-down" />
        </ButtonMove>
      </ButtonWrapper>
    </>
  );
}

const MovesWrapper = styled.section`
  background-color: ${props => props.theme.colors.greenScreen};
  padding: 10px 20px;
  border-radius: 3px;
  border: inset ${props => props.theme.colors.greenBorder} 3px;
  width: calc(100% - 100px);
`;

const MovementName = styled.h4`
  text-transform: capitalize;
  margin: 5px 0px;
`;

const MovementStatList = styled.ul`
  margin: 5px;
  border-top: 2px solid black;
  list-style-type: none;
  margin: 0px;
  padding: 0;
`;
const MovementStat = styled.li`
  font-weight: bold;
`;

const MoveInfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

const MoveTypeWrapper = styled.div``;
const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

const ButtonMove = styled.button`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  border: groove ${props => props.theme.colors.grey} 3px;
  margin: 5px;
`;
