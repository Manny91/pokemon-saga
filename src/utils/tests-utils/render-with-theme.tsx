import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
