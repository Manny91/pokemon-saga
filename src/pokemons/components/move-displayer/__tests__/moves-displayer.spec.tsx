import React from "react";
import { PokemonMoveDisplayer } from "../moves-displayer";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonMoveDisplayer Component", () => {
  it("should render properly when loading is false", () => {
    const testProps = {
      handleMoveNext: jest.fn(),
      handleMovePrevious: jest.fn(),
      move: {
        name: "razor-wind",
        pp: 10,
        power: 10,
        accuracy: 100,
        url: "test-url",
        id: 10,
        move: { name: "razor-wind", url: "test-url" }
      },
      loadingMoves: false
    };
    const subject = renderWithTheme(<PokemonMoveDisplayer {...testProps} />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });

  it("should render properly when loading is true", () => {
    const testProps = {
      handleMoveNext: jest.fn(),
      handleMovePrevious: jest.fn(),
      move: {
        name: "razor-wind",
        pp: 10,
        power: 10,
        accuracy: 100,
        url: "test-url",
        id: 10,
        move: { name: "razor-wind", url: "test-url" }
      },
      loadingMoves: true
    };
    const subject = renderWithTheme(<PokemonMoveDisplayer {...testProps} />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
