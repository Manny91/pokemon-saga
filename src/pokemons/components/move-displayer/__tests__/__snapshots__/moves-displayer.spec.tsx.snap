// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonMoveDisplayer Component should render properly when loading is false 1`] = `
.c0 {
  background-color: #A5CD53;
  padding: 10px 20px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
  width: calc(100% - 100px);
}

.c2 {
  text-transform: capitalize;
  margin: 5px 0px;
}

.c3 {
  margin: 5px;
  border-top: 2px solid black;
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.c4 {
  font-weight: bold;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  border: groove grey 3px;
  margin: 5px;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <h4
        class="c2"
      >
        razor-wind
      </h4>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          Power : 
          10
        </li>
        <li
          class="c4"
        >
          Accuracy : 
          100
        </li>
        <li
          class="c4"
        >
          PP : 
          10
        </li>
      </ul>
    </div>
    <div
      class=""
    />
  </section>
  <div
    class="c5"
  >
    <button
      class="c6"
    >
      <i
        class="fas fa-caret-up"
      />
    </button>
    <button
      class="c6"
    >
      <i
        class="fas fa-caret-down"
      />
    </button>
  </div>
</div>
`;

exports[`PokemonMoveDisplayer Component should render properly when loading is true 1`] = `
.c4 {
  width: 4px;
  height: 4px;
  background-color: black;
  border-radius: 1px;
  margin: 0px 2px;
  list-style-type: none;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  animation: fZpxSO 1s infinite ease-in-out both;
}

.c2 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c2 .c3:nth-child(1) {
  -webkit-animation-delay: -0.35s;
  animation-delay: -0.35s;
}

.c2 .c3:nth-child(2) {
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c0 {
  background-color: #A5CD53;
  padding: 10px 20px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
  width: calc(100% - 100px);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  border: groove grey 3px;
  margin: 5px;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3 c4"
        />
        <li
          class="c3 c4"
        />
        <li
          class="c3 c4"
        />
      </ul>
    </div>
    <div
      class=""
    />
  </section>
  <div
    class="c5"
  >
    <button
      class="c6"
    >
      <i
        class="fas fa-caret-up"
      />
    </button>
    <button
      class="c6"
    >
      <i
        class="fas fa-caret-down"
      />
    </button>
  </div>
</div>
`;
